### ISSUE TITLE
training parameters

### ISSUE THREAD

**Issue by YanShuo1992** [eric-yyjau/pytorch-superpoint#40]
Thanks for the work !

I am training my own network.  But I can not the same result as yours.
Could you please provide your training parameters?
Do you change you learning rate during training?  And do you use same img size 240*320 in export and training with COCO?

Cheers!

**Comment by eric-yyjau** [comment]
Hi @YanShuo1992 ,

Thanks for your message.
To generate keypoints for training, I suggest directly using the pre-trained network for homography adaptation (https://github.com/eric-yyjau/pytorch-superpoint/tree/master/logs/magicpoint_synth20/checkpoints).
For the parameters, you may refer to this one (https://github.com/eric-yyjau/pytorch-superpoint/blob/master/logs/superpoint_coco_heat2_0/config.yml)

I didn't change the learning rate. Yeah, image size should be 240*320.
Best,

**Comment by YanShuo1992** [comment]
Cheers @eric-yyjau 
I follow the same config and get a good result in my task.  However, I find an interesting phenomenon. 
When I train the model, the training and validation precision drop steeply.  Same thing happens in recall.
<img width="268" alt="valprecision" src="https://user-images.githubusercontent.com/26371531/112293863-90c5d280-8ccd-11eb-90f0-b97ecb539e24.png">

They will drop during about 5w-10w iters.  Then the precision climb to a reasonable value until 17w iters.

Did you find same phenomenon?  and any commits?

**Comment by eric-yyjau** [comment]
Hi @YanShuo1992 ,

Yeah I remember I saw this issue before but I don't know what caused this.
I think some pytorch version will have this issue.

If you know how to resolve this, please also let me know.
Thanks

**Comment by 35p32** [comment]
Thanks for your great work!  @eric-yyjau  and  really admire you。 
But there is something wrong during training  MagicPoint on Synthetic Shapes，I got the same problem as @YanShuo1992 。
I use  " torch1.9.0+cu111 "  and  the training precision and loss  looks strange:

![](https://ibb.co/vc5fNDN)
![](https://ibb.co/RTrjQmf)
![](https://ibb.co/f45sg4Q)
![](https://ibb.co/wK7GT2S)


Is this caused by the version? (if yes, What version do you recommend?) or something else ?

**Comment by eric-yyjau** [comment]
Hi @35p32,

Thanks for your question. Sorry I don't know why it happened.
Maybe it is because of the pytorch version. If you have a way to prevent this from happening, please share with me.
Thank you!

**Comment by johnnyboloni** [comment]
Hi,
I encountered the same problem. AMSGrad seems to have solved it for me:
optimizer = optim.Adam(net.parameters(), lr=lr, betas=(0.9, 0.999), amsgrad=True)

https://pytorch.org/docs/stable/generated/torch.optim.Adam.html#torch.optim.Adam

**Comment by XhqGlorry11** [comment]
@johnnyboloni hi, i'm trying to reproduce the result. Can you share your loss/metric curve and your software environment info please~thx

**Comment by johnnyboloni** [comment]
@XhqGlorry11 Hey, I'm using torch1.10.1+cu102 and python 3.6.13.
In orange is the original training, in red is after using AMSGrad. 
![image](https://user-images.githubusercontent.com/860333/186147199-5301c156-8d66-42f7-8491-0b18253907ae.png)

BTW, I'm now investigating why precision and recall so bad, even without the weird error:
![image](https://user-images.githubusercontent.com/860333/186147617-f3cba7ab-d3cb-4a6d-b7aa-f8672de48b09.png)

Do you get similar behavior?

**Comment by XhqGlorry11** [comment]
@johnnyboloni Thank you for your reply! I had a little problem in convergence. I want to re-confirm the training details with you.
I download [COCO pseudo ground truth](https://drive.google.com/drive/folders/1U2cXiWvngNqiLvnFsWHyRbMuQBvbAh_t) and train using [config](https://github.com/eric-yyjau/pytorch-superpoint/blob/master/configs/superpoint_coco_train_heatmap.yaml) with some modifications.
<img width="257" alt="image" src="https://user-images.githubusercontent.com/22702027/186149745-f97aef9a-1ad5-4d40-b430-6e143ed6157a.png">
<img width="626" alt="image" src="https://user-images.githubusercontent.com/22702027/186149857-c7ad06ff-f924-4db8-aa57-4b14af53d28a.png">
How about your settings?

**Comment by johnnyboloni** [comment]
@XhqGlorry11 I'm trying to train MagicPoint from scratch, I've used the SuperPointNet_gauss2 in the beginning, but now trying SuperPointNet. (I've used magicpoint_shapes_pair.yaml)

**Comment by XhqGlorry11** [comment]
@johnnyboloni I didn't train magic point from scratch. I directly use COCO pseudo ground truth and train SuperPointNet. Here is my metric curve.
<img width="351" alt="image" src="https://user-images.githubusercontent.com/22702027/186311085-2df4ddb4-7df2-45c2-a3a2-910e7979613b.png">
<img width="355" alt="image" src="https://user-images.githubusercontent.com/22702027/186311103-ee63e66a-557d-4b83-a778-e70c29362e06.png">
<img width="355" alt="image" src="https://user-images.githubusercontent.com/22702027/186311120-86df6d54-cd8f-47ca-b366-6da262f5ccd4.png">

**Comment by FeiXie8** [comment]
> @johnnyboloni I didn't train magic point from scratch. I directly use COCO pseudo ground truth and train SuperPointNet. Here is my metric curve. <img alt="image" width="351" src="https://user-images.githubusercontent.com/22702027/186311085-2df4ddb4-7df2-45c2-a3a2-910e7979613b.png"> <img alt="image" width="355" src="https://user-images.githubusercontent.com/22702027/186311103-ee63e66a-557d-4b83-a778-e70c29362e06.png"> <img alt="image" width="355" src="https://user-images.githubusercontent.com/22702027/186311120-86df6d54-cd8f-47ca-b366-6da262f5ccd4.png">

is this trannning data in tensorborad is as good as you expect?
